### PR TITLE
combine settings and list styles

### PR DESCRIPTION
**Pull request by donni106**
This PR introduces the functionality for individual list types per settings.
It is based on https://github.com/ikuseiGmbH/smart-village-app-app/pull/119.
Everything should work directly after checkout of the branch, there is no additional work required.
Please see commit messages for further descriptions.

sva-list-types

### CODE REVIEW

File: src/components/SettingsListItem.js [hunk 1]
```diff
@@ -1,72 +1,119 @@
...
+          <RegularText small></RegularText>
+          {Object.values(LIST_TYPES).map((listType) => (
+            <Radiobutton
+              key={listType}
+              title={texts.settingsTitles.listLayouts[listType]}
+              onPress={async () => {
+                if (listType !== selectedListType) {
+                  setSelectedListType(listType);
+                  // call the onPress callback for that setting list item if present
+                  onPress && (await onPress(listType));
```
Comment (digorath) [inline src/components/SettingsListItem.js:98]: as far as I can tell the onPress is not `async`. can you double check?

File: src/screens/SettingsScreen.js [hunk 2]
```diff
@@ -54,6 +53,18 @@ export const SettingsScreen = ({ navigation }) => {
     }, 500);
   };
 
+  const onPress = async (selectedListType, queryType) =>
```
Comment (digorath) [inline src/screens/SettingsScreen.js:56]: I think a more specific name than `onPress` inside of a component with many things that can be pressed/touched would be appropriate

**Review by digorath — CHANGES_REQUESTED** [review]

File: src/GlobalSettingsProvider.js [hunk 3]
```diff
@@ -1,15 +1,27 @@
...
+import React, { createContext, useState } from 'react';
 
 export const GlobalSettingsContext = createContext({});
 
-export const GlobalSettingsProvider = ({ globalSettings, children }) => (
-  <GlobalSettingsContext.Provider value={{ ...globalSettings }}>
-    {children}
-  </GlobalSettingsContext.Provider>
-);
+export const GlobalSettingsProvider = ({
```
[the remark this answers is not in the document]
Comment (donni106) [reply src/GlobalSettingsProvider.js:6]: good point

File: src/screens/SettingsScreen.js (hunk 2 above)
Comment (donni106) [reply src/screens/SettingsScreen.js:56]: agree

Comment (donni106) [reply src/screens/SettingsScreen.js]: done in https://github.com/ikuseiGmbH/smart-village-app-app/pull/129/commits/552c906697f61fb7cb7d322ae6366ce47f5de9e1

File: src/components/SettingsListItem.js (hunk 1 above)
Comment (donni106) [reply src/components/SettingsListItem.js]: done in https://github.com/ikuseiGmbH/smart-village-app-app/pull/129/commits/87ace9b611e8293292a8df444c876d555f1258a0

File: src/GlobalSettingsProvider.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (donni106) [reply src/GlobalSettingsProvider.js]: done in https://github.com/ikuseiGmbH/smart-village-app-app/pull/129/commits/22c851eed4f312a43b7e5733d9fd1d013eb7d0e3

**Review by digorath — APPROVED** [review]

**Pull request merged**